### DISCUSSION TITLE
OPCUA Errors

### DISCUSSION THREAD

**Discussion by AndreyKl4** [FreeOpcUa/python-opcua#1223] [Q&A]
Hi All,

can smb please explain me  in what way is correct to represent Errors in OPCUA Methods. For Example:

I have two InputArguments x,y and in Server Method callback i want to divide them z = x/y and I want to raise opcua error in Client if InputArgument y=0 and Method Call should not be succeeded. Unfortunately, i didn't find a way to declare opcua erros in method callback....

Thanks in Advance!

Andrey

**Comment by AndreasHeine** [comment]
https://reference.opcfoundation.org/v104/Core/docs/Part4/5.11.2/

> StatusCode   of the Method executed in the server. This StatusCode is set to the Bad_InvalidArgument StatusCode if at least one input argument broke a constraint (e.g. wrong data type, value out of range).
> This StatusCode is set to a bad StatusCode if the Method execution failed in the server, e.g. based on an exception.

`return ua.StatusCode(ua.status_codes.StatusCodes.BadInvalidArgument)`

you can also return a instance of "CallMethodResult":
https://github.com/FreeOpcUa/python-opcua/blob/1cca33dd5fb70858b5e811b08356ca97520bcbaf/opcua/common/methods.py#L72
https://github.com/FreeOpcUa/python-opcua/blob/1cca33dd5fb70858b5e811b08356ca97520bcbaf/opcua/ua/uaprotocol_auto.py#L8316

**Comment by oroulet** [comment]
The main method is supposed to be to raise the UaError

`raise ua.errors.BadInvalidArgument` or similar

**Reply by AndreyKl4** [reply]
@oroulet @AndreasHeine  Thanks a lot for the quick and very helpful answers!!!
